deep_field_metadetect/gaussmom/gaussmom.py

```diff
@@ -0,0 +1,417 @@
+import jax
+import jax.numpy as jnp
+import ngmix
+import numpy as np
+from ngmix.moments import MOMENTS_NAME_MAP
+
+from deep_field_metadetect.gaussmom.gaussmom_core import (
+    GaussMomData,
+    GaussMomObs,
+    _eval_gauss2d,
+)
+
+jax.config.update("jax_enable_x64", True)
+
+
+@jax.jit
+def fwhm_to_sigma(fwhm: float):
+    """
+    convert fwhm to sigma for a gaussian
+    """
+    return fwhm / 2.3548200450309493  # sig = fwhm / sqrt(8 * ln 2)
+
+
+@jax.jit
+def fwhm_to_T(fwhm):
+    """
+    convert fwhm to T for a gaussian
+    """
+    sigma = fwhm_to_sigma(fwhm)
+    return 2 * sigma**2
+
+
+@jax.jit
+def create_circular_mask(umod, vmod, maxrad):
+    distance_from_center = np.sqrt((umod) ** 2 + (vmod) ** 2)
+
+    mask = jnp.where(distance_from_center <= maxrad, 1, 0)
+
+    return mask.astype(int)
+
+
+def get_weighted_moments(obs, T, maxrad=None):
+    """
+    Get weighted moments using this mixture as the weight, including
+    e1,e2,T,s2n etc.  If you just want the raw moments use
+    get_weighted_sums()
+
+    If you want the expected fluxes, you should set the flux to the inverse
+    of the normalization which is 2*pi*sqrt(det)
+
+    Parameters
+    ----------
+    obs: Observation
+        The Observation to compare with. See ngmix.observation.Observation
+        The Observation must have a weight map set
+    maxrad: float, optional
+        If sent, limit moments to within the specified maximum radius
+    with_higher_order: bool, optional
+        If set to True, return higher order moments in the sums/sums_cov
+        arrays.  See ngmix.moments.MOMENTS_NAME_MAP for a map between
+        name and index.
+
+    Returns
+    -------
+    result array with basic sums as well as summary statistics
+    such as e1,e2,T,s2n etc.
+    """
+
+    sums, sums_cov, wsum, _ = get_weighted_sums(
+        obs,
+        T=T,
+        maxrad=maxrad,
+    )
+    return get_weighted_moments_stats(obs, sums, sums_cov, wsum)
+
+
+# @jax.jit
```

Comment:
Is there a reason the jit decorators are commented out? Do we want to have it by default? I am not sure, but I am leaning towards yes.

Reply:
it was just for debugging, jitting is definitely making it faster